packages/components/ng-ovh-uirouter-layout/src/index.js

```diff
@@ -0,0 +1,205 @@
+import intersection from 'lodash/intersection';
+import isString from 'lodash/isString';
+import kebabCase from 'lodash/kebabCase';
+import last from 'lodash/last';
+import map from 'lodash/map';
+import reduce from 'lodash/reduce';
+import set from 'lodash/set';
+import size from 'lodash/size';
+import startsWith from 'lodash/startsWith';
+import xor from 'lodash/xor';
+
+const moduleName = 'ngOvhUiRouterLayout';
+
+angular
+  .module(moduleName, [
+    'ui.bootstrap',
+    'ui.router',
+  ])
+  .config(/* @ngInject */($urlServiceProvider, $stateProvider, $transitionsProvider, $injector) => {
+    /* ========================================================
+    =            UI ROUTER MODAL LAYOUT DEFINITION            =
+    ========================================================= */
+
+    let modalInstance = null;
+
+    /**
+     * Delay initial URL synchronization.
+     * As we will create some states later (at run phase), maybe the url we want to access
+     * is not yet configured. So wait a moment that all states are declared (at config phase)
+     * and re-launch the URL sync later.
+     */
+    $urlServiceProvider.deferIntercept();
+
+    /**
+     *  Create a decorator for our new state attribute 'layout'.
+     *  For modal layout, the attribute can be a string or an object:
+     *  - if string - value must be 'modal'
+     *  - if object - avaiable attributes are:
+     *    - name: the value must be 'modal'.
+     *    - toChilds (default value: 'false'):
+     *      - can be a boolean: 'true' to declare the modal state to all direct parent childs.
+     *      - or an array of string that contains the childs of the direct parents where the modal
+     *        needs to be displayed.
+     *    - ignoreChilds: an array of string that contains states names where the modal state
+     *                    doesn't need to be displayed.
+     */
+    $stateProvider.decorator('layout', (state) => {
+      let modalLayout;
+      if (get(state, 'self.layout') === 'modal' || get(state, 'self.layout.name') === 'modal') {
+        modalLayout = {
+          name: 'modal',
+          toChilds: state.self.layout.toChilds || false,
+          ignoreChilds: state.self.layout.ignoreChilds || [],
+          redirectTo: state.self.layout.redirectTo || '^',
+        };
+      }
+
+      return modalLayout;
+    });
+
+    /**
+     * Use onSuccess hook to manage the modal display.
+     */
+    $transitionsProvider.onSuccess({}, (transition) => {
+      transition.promise.finally(() => {
+        const state = transition.$to();
+
+        // close previous modal
+        if (modalInstance) {
+          modalInstance.close();
+        }
+
+        if (get(state, 'layout.name') === 'modal') {
+          const $state = transition.injector().get('$state');
+          const $uibModal = transition.injector().get('$uibModal');
+
+          const componentName = get(state, state.views.modal ? 'views.modal.component' : 'component');
```

Comment:
I don't understand why we need to check `state.views.modal`. Why do we not use the default view ?

Reply:
You could define state like this : 

```js
$stateProvider
  .state('project.storage_blocks', {
    url: '/storages/blocks',
    layout: 'modal',
    views: {
        '@main': {
          component: 'myAwesomeComponent',
        },
        modal: {
          component: 'myModalComponent',
        },
    },
  );
```

the `views.modal` will open in modal.

Reply:
Strange no ? If you open a modal, why do we want to display another view in the background ?